src/runtime/swg-client-event-manager.js

```diff
@@ -0,0 +1,312 @@
+  }
+
+  /**
+   * @param {AnalyticsEvent} value
+   */
+  setEventType(value) {
+    assert(SwgClientEvent.isValidType_(value),
+        'Invalid event type in SwgClientEvent (' + value + ')');
+    this.eventType_ = value;
+  }
+
+  /** True if the user took direct action to initiate this event.  False if the
+   *  event was generated passively in the background.
+   * @return {boolean|null}
+   */
+  getIsFromUserAction() {
+    return this.isFromUserAction_;
+  }
+
+  /**True if the user took direct action to initiate this event.  False if the
+   * event was generated passively in the background.  Leave this value null
+   * if you don't know the correct answer.
+   * @param {boolean|null} value
+   */
+  setIsFromUserAction(value) {
+    assert(SwgClientEvent.isValidIsFromUserAction_(value),
+        'Invalid isFromUserAction in SwgClientEvent (' + value + ')');
+    this.isFromUserAction_ = value;
+  }
+
+  /**An optional parameter bag that can be included to provide additional
+   * details about some events.
+   * @return {Object}
+   */
+  getAdditionalParameters() {
+    return this.additionalParameters_;
+  }
+
+  /**An optional parameter bag that can be included to provide additional
+   * details about some events.  If you want to add or override a property,
+   * use the get method to get a reference to the existing object.
+   * @param {Object} value
+   */
+  setAdditionalParameters(value) {
+    assert(SwgClientEvent.isValidAdditionalParameters_(value),
+        'Invalid additionalParameters in SwgClientEvent (' + value + ')');
+    this.additionalParameters_ = value;
+  }
+
+  /**Returns true if the object is valid.  Otherwise it throws an error if you
+   * pass true and returns false if you don't.
+   * @param {boolean} createError
+   */
+  isValid(createError) {
+    if (!SwgClientEvent.isValidType_(this.eventType_)) {
+      if (!createError) {
+        return false;
+      }
+      throw new Error('Invalid event type in SwgClientEvent (' +
+          this.eventType_ + ')');
+    }
+    if (!SwgClientEvent.isValidOriginator_(this.eventOriginator_)) {
+      if (!createError) {
+        return false;
+      }
+      throw new Error('Invalid event originator in SwgClientEvent (' +
+          this.eventOriginator_ + ')');
+    }
+    if (!SwgClientEvent.isValidIsFromUserAction_(this.isFromUserAction_)) {
+      if (!createError) {
+        return false;
+      }
+      throw new Error('Invalid isFromUserAction in SwgClientEvent (' +
+          this.isFromUserAction_ + ')');
+    }
+    if (!SwgClientEvent.isValidAdditionalParameters_(
+        this.additionalParameters_)) {
+      if (!createError) {
+        return false;
+      }
+      throw new Error('Invalid additional parameters in SwgClientEvent (' +
+          this.additionalParameters_ + ')');
+    }
+    return true;
+  }
+}
+
+export class SwgClientEventManager {
+
+  /**True if the value is a function
+   * @param {*} value
+   * @private
```

Comment:
Since this is not specific to this class,maybe add this if not already present into utils